### ISSUE TITLE
Order songs by file name when browsing folders via Subsonic

### ISSUE THREAD

**Issue by grisumedia** [owncloud/music#1182]
I'd like to use the Android app Symfonium using the Subsonic API. Unfortunately, the files in the folders are always ordered by their title. Symfonium does not allow to change the file order. As I understand from [a ticket in the Symfonium support forum](https://support.symfonium.app/t/default-sorting-on-files-view-subsonic/2360), the order comes from the server.
I would expect that the files are in the order of their name, not ordered by title.
Is it possible to fix that or to make that configurable?

![Screenshot: Order in Symfonium](https://github.com/user-attachments/assets/0d258256-c7d7-48a8-987c-ac0307e2b053)
![Screenshot: File names](https://github.com/user-attachments/assets/2322b234-7c89-4c82-a5fa-c6e56fcf5af4)

**Comment by grisumedia** [comment]
I agree, different sortings in different views completely make sense. And, if it's possible, it would maybe also be nice so see the file names instead of the title when browsing by folders.

**Comment by paulijar** [comment]
The sorting change is now released in Music v2.1.0. 

Showing file names instead titles would not be possible without dirty hack of pretending that the file name is actually the title. This could cause other problems for some clients (*). However, I now include the file name and path in the property `path` of the returned songs. I had previously postponed implementing this since this property is specified as optional and efficient creation of the full path was quite tricky. Hence, if the client wishes to show the file names, it's now technically possible.

(*) In the past, I used to have kind of context-specific track numbers on the song objects returned in the Subsonic API. However, it turned out that it distracted the caching logic of some clients when the same song entry could have different data depending on the context. This title/filename thing would probably have similar effect.
